types/types.go

```diff
@@ -38,7 +38,16 @@ type SignatureEventSelector struct {
 type SignatureHandler func(found Finding)
 
 //Event is a generic event that the Engine can process
-type Event interface{}
+type EventHeader struct {
```

Comment:
I'm okey with the current structure, but maybe we could rename EventHeader to EventHeaders to indicate that we can have multiple headers. I'd make it even more generic with helper methods to retrieve Content Type and Origin:

```go
type EventHeaders map[string]string

type Event struct {
	Headers  EventHeaders
	Payload interface{}
}

func (e Event) ContentType() string {
	return e.Headers["Content-Type"]
}

func (e Event) Origin() string {
	return e.Headers["Origin"]
}
```

Reply:
Perhaps we could have the headers fulfill such an interface?
```
type EventHeaders interface {
    ContentType() string
    Origin()             string
}
```

Like this we could indicate "mandatory" header results.

Reply:
In my code snippet the Event stuct fulfils such interface. Structs and methods is probably all we need for now.

Reply:
I think CloudEvents do the same thing with defining their headers as an interface (They do that so they can version their header structs). 
However, I am also slightly worried with using `map[string]string` for headers because we will need to allocate a map for each event pushed to tracee-rules, this can get expensive overtime (and performance is one of the reasons for this refactor as well from my PoV).

Reply:
Agree. So maybe what you proposed is a good tradeoff:

```go
type EventHeaders struct {
	// ContentType indicates the content of the Event payload.
	ContentType string
	// Origin indicates whether Event originates from host or container.
	Origin string
	
	// Headers additional custom headers, nil most of the time
	Headers map[string]string
}

type Event struct {
	Headers EventHeaders
	
	Payload interface{}
}

func (e Event) ContentType() string {
	return e.Headers.ContentType
}

func (e Event) Origin() string {
	return e.Headers.Origin
}
```

Reply:
Done